src/sentry/utils/flag.py

```diff
@@ -0,0 +1,80 @@
+# Thread safe flag tracking wrapper.
+from contextvars import ContextVar
+
+flag_manager = ContextVar("flag_manager")
+
+
+def initialize_flag_manager(capacity: int = 10) -> None:
+    flag_manager.set(FlagManager(capacity=capacity))
+
+
+# NOTE: If not properly initialized this function is a no-op.
+def process_flag_result(flag: str, result: bool) -> None:
+    try:
+        _flag_manager = flag_manager.get()
+        _flag_manager.insert(flag, result)
+    except LookupError:
```

Comment:
What can throw LookupError? `flag_manager.get()`?
would we want a debug level log to figure out if things were not properly initialized?

Reply:
flag_manager.get() is the only location that throws this error.  I'm not logging at the moment because I know there are gaps and I know this will be noisy.

Reply:
What are the known gaps? If that's known/obvious to everyone feel free to ignore this since I'm catching up

Reply:
Known only to me.  I'm initializing the variable for the HTTP server but not the tasks or consumers.

Reply:
Is that worth writing in the PR description or as comment somewhere?